src/mscorlib/shared/System/MemoryExtensions.Fast.cs

```diff
@@ -359,6 +359,7 @@ public static bool StartsWith(this ReadOnlySpan<char> span, ReadOnlySpan<char> v
         /// <summary>
         /// Creates a new span over the portion of the target array.
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Span<T> AsSpan<T>(this T[] array, int start)
         {
             if (array == null)
```

Comment:
Orthogonal to this PR: We should consider creating an internal static Create method on Span<T> (in Span.Fast.cs) just like the portable span and move this AsSpan method into MemoryExtensions.cs (since the implementation between the two would be the same.
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Span.Portable.cs#L48
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/MemoryExtensions.Portable.cs#L20

Reply:
I do not think this is a good idea. These extra layers makes the JIT work harder and tend to leave inefficiencies behind. We should not be de-optimizing fast Span just so we can share a few more lines with the slow Span.